azure-iot-device/azure/iot/device/common/pipeline/pipeline_stages_base.py

```diff
@@ -793,11 +673,8 @@ def _execute_op(self, op):
         Send all ops down and intercept their return to "watch for retry"
         """
         if self._should_watch_for_retry(op):
-            self.send_op_down_and_intercept_return(
-                op=op, intercepted_return=self._on_intercepted_return
-            )
-        else:
-            self.send_op_down(op)
+            op.add_callback(self._do_retry_if_necessary)
```

Comment:
>_do_retry_if_necessary [](start = 33, length = 22)

This is great.  I like that the name of the callback describes what it does.  Huge improvement.